src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5781,7 +5781,8 @@ void makeAndAdjustReceiverSlot(BoundExpression receiver)
                 Join(ref this.State, ref consequenceState);
 
                 TypeSymbol? refResultType = node.Type?.SetUnknownNullabilityForReferenceTypes();
-                if (IsNullabilityMismatch(consequenceLValue, alternativeLValue))
+                if (!originalConsequence.IsSuppressed && !originalAlternative.IsSuppressed &&
+                    IsNullabilityMismatch(consequenceLValue, alternativeLValue))
```

Comment:
Taking a look at other locations that use `IsNullabilityMismatch`, it looks like there's an other one with a similar problem (doesn't first check for suppression), in `VisitForEachIterationVariables`.
The scenario is `foreach (ref object o in refCollectionOfNullableObjects!)`. More details below.
Feels like that should work. But since that problem was not reported by any users yet,  I'm fine to punt. 
Just wanted to mention it.

```
#nullable enable

using System;
using System.Collections;
using System.Collections.Generic;

public class RefList : IEnumerable<object>
{
    private object[] _items;

    public RefList(object[] items)
    {
        _items = items;
    }

    public RefEnumerator GetEnumerator() => new RefEnumerator(_items);

    IEnumerator<object> IEnumerable<object>.GetEnumerator() => GetEnumerator();

    IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();

    public struct RefEnumerator : IEnumerator<object?>
    {
        private object?[] _items;
        private int _index;

        public RefEnumerator(object?[] items)
        {
            _items = items;
            _index = -1;
        }

        public ref object? Current => ref _items[_index];

        object? IEnumerator<object?>.Current => Current;

        object? IEnumerator.Current => Current;

        public bool MoveNext()
        {
            _index++;
            return _index < _items.Length;
        }

        public void Reset() => _index = -1;

        public void Dispose() { }
    }
}

class Program
{
    static void M(RefList refList)
    {
        foreach (ref object item in refList!) // suppression doesn't affect warning
        {
        }
    }
}

```

Reply:
That's a good catch, thanks. I can try to fix that as part of this PR too.